src/Stripe.net/Infrastructure/JsonConverters/STJStripeObjectConverter.cs

```diff
@@ -0,0 +1,129 @@
+    /// decide which concrete type to instantiate. If no concrete type is found (or if one is found,
+    /// but it's not compatible with the expected interface), then the converter returns `null`.
+    /// </summary>
+    ///
+    public class STJStripeObjectConverter : JsonConverterFactory
+    {
+        /// <summary>
+        /// Determines whether this instance can convert the specified object type.
+        /// </summary>
+        /// <param name="objectType">Type of the object.</param>
+        /// <returns>
+        ///     <c>true</c> if this instance can convert the specified object type; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool CanConvert(Type objectType)
+        {
+            var typeInfo = objectType.GetTypeInfo();
+            return typeInfo.IsInterface && typeInfo.FullName.StartsWith("Stripe");
+        }
+
+        public override JsonConverter CreateConverter(
+            Type type,
+            JsonSerializerOptions options)
+        {
+#pragma warning disable SA1009 // Closing parenthesis should be spaced correctly
+            JsonConverter converter = (JsonConverter)Activator.CreateInstance(
+                typeof(STJStripeObjectConverterInner<>).MakeGenericType(
+                    type),
+                BindingFlags.Instance | BindingFlags.Public,
+                binder: null,
+                args: null,
+                culture: null)!;
+#pragma warning restore SA1009 // Closing parenthesis should be spaced correctly
+
+            return converter;
+        }
+
+        internal class STJStripeObjectConverterInner<T> : JsonConverter<T>
+        where T : class
+        {
+            /// <summary>
+            /// Reads the JSON representation of the object.
+            /// </summary>
+            /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+            /// <param name="typeToConvert">Type of the object.</param>
+            /// <param name="options">The calling serializer's options.</param>
+            /// <returns>The object value.</returns>
+            public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    return null;
+                }
+
+                var jsonObject = JsonElement.ParseValue(ref reader);
+                var objectValue = jsonObject.GetProperty("object").GetString();
+
+                Type concreteType = StripeTypeRegistry.GetConcreteType(typeToConvert, objectValue);
+                if (concreteType == null)
+                {
+                    // Couldn't find a concrete type to instantiate, return null.
+                    return null;
+                }
+
+                return (T)JsonSerializer.Deserialize(jsonObject.GetRawText(), concreteType, options);
```

Comment:
can we go through reader / element overload?

Reply:
This was the closest I could find to what we currently do in NS but I think we may be able to make this more efficient with a little work.

Reply:
Update: there is an override that I initially didn't notice that lets us pass jsonObject itself without going thru raw text